pkg/handlers/embedded_cluster_node_join_command.go

```diff
@@ -0,0 +1,116 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/replicatedhq/kots/pkg/embeddedcluster"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/store/kotsstore"
+)
+
+type GenerateEmbeddedClusterNodeJoinCommandResponse struct {
+	Command []string `json:"command"`
+}
+
+type GetEmbeddedClusterNodeJoinCommandResponse struct {
+	ClusterID      string `json:"clusterID"`
+	K0sJoinCommand string `json:"k0sJoinCommand"`
+	K0sToken       string `json:"k0sToken"`
+}
+
+type GenerateEmbeddedClusterNodeJoinCommandRequest struct {
+	Roles []string `json:"roles"`
+}
+
+func (h *Handler) GenerateEmbeddedClusterNodeJoinCommand(w http.ResponseWriter, r *http.Request) {
+	generateEmbeddedClusterNodeJoinCommandRequest := GenerateEmbeddedClusterNodeJoinCommandRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&generateEmbeddedClusterNodeJoinCommandRequest); err != nil {
+		logger.Error(fmt.Errorf("failed to decode request body: %w", err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	store := kotsstore.StoreFromEnv()
+	token, err := store.SetEmbeddedClusterInstallCommandRoles(generateEmbeddedClusterNodeJoinCommandRequest.Roles)
```

Comment:
And I'd delete the line above
```suggestion
	token, err := store.GetStore().SetEmbeddedClusterInstallCommandRoles(generateEmbeddedClusterNodeJoinCommandRequest.Roles)
```